ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -391,11 +377,15 @@ private boolean deleteIfNeeded(BlobId blobId, long deletionTime, short lifeVersi
     // Note: always check cache before operation attempt, since this could be a retry after a CONFLICT error,
     // in which case the cache may have been updated by another thread.
     if (!checkCacheState(blobKey, lifeVersion, BlobState.DELETED)) {
-      boolean deleted = cloudDestination.deleteBlob(blobId, deletionTime, lifeVersion);
+      boolean deleted = cloudDestination.deleteBlob(blobId, deletionTime, lifeVersion,
+          (metadata, key, updateFields) -> preDeleteValidation(metadata, key, updateFields));
       addToCache(blobKey, lifeVersion, BlobState.DELETED);
       return deleted;
+    } else {
+      throw new CloudStorageException("Error updating blob metadata",
```

Comment:
We might need to distinguish between vcr and live traffic.  For VCR, shouldUpload() can return false if the TTL is low.